clippy_utils/src/sugg.rs

```diff
@@ -33,7 +33,7 @@ pub enum Sugg<'a> {
     MaybeParen(Cow<'a, str>),
     /// A binary operator expression, including `as`-casts and explicit type
     /// coercion.
-    BinOp(AssocOp, Cow<'a, str>),
+    BinOp(AssocOp, (Cow<'a, str>, Cow<'a, str>), Cow<'a, str>),
```

Comment:
I don't think the last snippet here is needed.

```suggestion
    BinOp(AssocOp, Cow<'a, str>, Cow<'a, str>),
```

Reply:
I rewrite it.